crates/pipeline_manager/src/db/mod.rs

```diff
@@ -1385,4 +1510,80 @@ impl ProjectDB {
         txn.commit().await?;
         Ok(())
     }
+
+    /// Retrieves the service identifier using its name, and uses this
+    /// identifier in the creation of a new service probe.
+    pub async fn new_service_probe_with_service_name(
+        &self,
+        tenant_id: TenantId,
+        service_name: &str,
+        id: Uuid,
+        request: ServiceProbeRequest,
+        created_at: &DateTime<Utc>,
+    ) -> Result<ServiceProbeId, DBError> {
+        let mut manager = self.pool.get().await?;
+        let txn = manager.transaction().await?;
+        let service = self
+            .get_service_by_name(tenant_id, service_name, Some(&txn))
+            .await?;
+        let service_probe_id = self
+            .new_service_probe(
+                tenant_id,
+                service.service_id,
+                id,
+                request,
+                created_at,
+                Some(&txn),
+            )
+            .await?;
+        txn.commit().await?;
+        Ok(service_probe_id)
+    }
+
+    /// Retrieves the service identifier using its name, and uses this
+    /// identifier to fetch the specific service probe.
+    pub(crate) async fn get_service_probe_with_service_name(
```

Comment:
Could simplify to `get_service_probe()` (same comment for the other methods here)

Reply:
This would lead to a naming conflict with `get_service_probe(id)` (same for the other comments) as Rust does not allow function overloading. I could use [fully qualified syntax](https://doc.rust-lang.org/stable/book/ch19-03-advanced-traits.html#fully-qualified-syntax-for-disambiguation-calling-methods-with-the-same-name) to get around it, though I'm more in favor of keeping the longer function names as they are descriptive of what it does (retrieve the service probe ensuring that the service name is correct, or for list_probes the service name is more than a sanity check).

The reason I didn't simply have the service functions directly use names in their queries is such that database proptesting still works without a very large amount of edits, and to not deviate from the pattern already used for the other object types (program, connector, pipeline).

Reply:
Makes sense. If there is `get_service_probe(id)`, then a minor improvement for the by-name variant could be `get_service_probe_for_service(service_name)`. The fact that the argument is called `service_name` makes it clear what the fetching parameter is.

Reply:
Changed to:
- `new_service_probe_for_service_by_name`
- `get_service_probe_for_service_by_name`
- `list_service_probes_for_service_by_name`

The `for_service` part is there such that the `by_name` part is about the service name rather than a service probe having a new. I considered `for_named_service` but all services are named, as such it might be confusing.

Reply:
You could do "xyz_probe_for_service..." and skip the first "service". :)